File: embryo_dev_temperature_solve.py
# enter desired final NF-stage here:
nfStage = 34

# enter hours until that stage should be reached:
time = 48+8

import numpy as np
from scipy.optimize import root_scalar

if nfStage <= 40:
    f1 = lambda temp: -137.4*np.exp(-0.1460*temp) + (21.54*np.exp(-0.1192*temp))*nfStage-time
else:
    f1 = lambda temp: -2481.0*np.exp(-0.09332*temp) + (81.97*np.exp(-0.09745*temp))*nfStage-time

temperature = round((root_scalar(f1, bracket=[0, 35], method='brentq')).root, 1)

print(f"Incubate at {temperature}°C")

File: embryo_dev_temperature_solve.R


#enter desired final NF-stage here:
nfStage <- 40.1

#enter hours until that stage should be reached:
time <- 60



library("rootSolve")

if(nfStage<=40){
  f1 = function(temp){
    -137.4*exp(-0.1460*temp) + (21.54*exp(-0.1192*temp))*nfStage-time
  }
} else {
  f1 = function(temp){
    -2481.0*exp(-0.09332*temp) + (81.97*exp(-0.09745*temp))*nfStage-time
  }
}
temperature = round((uniroot.all(f1, c(0,35))), digits = 1)

print(paste("Incubate at", temperature, "°C"))


